layout/new_test.go

```diff
@@ -0,0 +1,237 @@
+						h.AssertEq(t, attribute, "linux")
+
+						// #Variant
+						attribute, err = idx.Variant(linuxAmd64Digest)
+						h.AssertNil(t, err)
+						h.AssertEq(t, attribute, "v1")
+
+						// #OSVersion
+						attribute, err = idx.OSVersion(linuxAmd64Digest)
+						h.AssertNil(t, err)
+						h.AssertEq(t, attribute, "4.5.6")
+
+						// #OSFeatures
+						attributes, err = idx.OSFeatures(linuxAmd64Digest)
+						h.AssertNil(t, err)
+						h.AssertContains(t, attributes, "os-feature-1", "os-feature-2")
+
+						// #Features
+						attributes, err = idx.Features(linuxAmd64Digest)
+						h.AssertNil(t, err)
+						h.AssertContains(t, attributes, "feature-1", "feature-2")
+					})
+				})
+
+				when("linux/arm64", func() {
+					it("attributes are readable", func() {
+						// #Architecture
+						attribute, err = idx.Architecture(linuxArm64Digest)
+						h.AssertNil(t, err)
+						h.AssertEq(t, attribute, "arm")
+
+						// #OS
+						attribute, err = idx.OS(linuxArm64Digest)
+						h.AssertNil(t, err)
+						h.AssertEq(t, attribute, "linux")
+
+						// #Variant
+						attribute, err = idx.Variant(linuxArm64Digest)
+						h.AssertNil(t, err)
+						h.AssertEq(t, attribute, "v7")
+
+						// #OSVersion
+						attribute, err = idx.OSVersion(linuxArm64Digest)
+						h.AssertNil(t, err)
+						h.AssertEq(t, attribute, "1.2.3")
+
+						// #OSFeatures
+						attributes, err = idx.OSFeatures(linuxArm64Digest)
+						h.AssertNil(t, err)
+						h.AssertContains(t, attributes, "os-feature-3", "os-feature-4")
+
+						// #Features
+						attributes, err = idx.Features(linuxArm64Digest)
+						h.AssertNil(t, err)
+						h.AssertContains(t, attributes, "feature-3", "feature-4")
+					})
+				})
+			})
+
+			when("#Annotations", func() {
+				var annotations map[string]string
+
+				when("linux/amd64", func() {
+					it("existing annotations are readable", func() {
+						annotations, err = idx.Annotations(linuxAmd64Digest)
+						h.AssertNil(t, err)
+						h.AssertEq(t, annotations["com.docker.official-images.bashbrew.arch"], "amd64")
+						h.AssertEq(t, annotations["org.opencontainers.image.url"], "https://hub.docker.com/_/busybox")
+						h.AssertEq(t, annotations["org.opencontainers.image.revision"], "d0b7d566eb4f1fa9933984e6fc04ab11f08f4592")
+					})
+				})
+
+				when("linux/arm64", func() {
+					it("existing annotations are readable", func() {
+						annotations, err = idx.Annotations(linuxArm64Digest)
+						h.AssertNil(t, err)
+						h.AssertEq(t, annotations["com.docker.official-images.bashbrew.arch"], "arm32v7")
+						h.AssertEq(t, annotations["org.opencontainers.image.url"], "https://hub.docker.com/_/busybox")
+						h.AssertEq(t, annotations["org.opencontainers.image.revision"], "185a3f7f21c307b15ef99b7088b228f004ff5f11")
+					})
+				})
+			})
+
+			when("#URLs", func() {
+				when("linux/amd64", func() {
+					it("existing annotations are readable", func() {
+						t.Skip("Do we really want to support this now???, its failing")
```

Comment:
Maybe I didn't set the `subject` data correctly, but I didn't know how to test this. Maybe we can remove that method, honestly, I do not thing the community is asking for it and it adds more complexity to the code

Reply:
I personally felt the same, But i saw similar implementation in podman to Set and Get URLs, I think it is using those setters and getters internally for something else.

Currently, the setters like `SetFeatures`, `SetOSFeatures`, `SetAnnotations` actually append data instead of replacing(similar to Podman and Docker), I wonder **what if the CNB author wants to remove any feature, osFeature or annotation**?

At present I think it is not a priority, incase if its helpful to CNB authors I propose adding `features`, `osFeatures` and `annotations` flags to `pack manifest rm` command

- when `pack manifest rm < some-image | some-index >`, the image or index with given digest will be removed
- when `pack manifest rm < some-image | some-index > [ --features | --os-features | --annotations ]` then instead of removing image, pack removes features, osFeatures and annotations from specified image.